internal/config/file.go

```diff
@@ -19,8 +24,58 @@ func (f File) Read() (string, error) {
 	return string(byt), err
 }
 
-// Coder CLI configuration files.
-var (
-	Session File = "session"
-	URL     File = "url"
+// UnmarshalYAML reads the file and unmarshals as yaml.
+func (f File) UnmarshalYAML(out interface{}) error {
+	byt, err := read(string(f))
+	if err != nil {
+		return xerrors.Errorf("read file: %w", err)
+	}
+	if err := yaml.Unmarshal(byt, out); err != nil {
+		return xerrors.Errorf("unmarshal yaml: %w", err)
+	}
+	return nil
+}
+
+// WriteYAML writes the file as yaml.
+func (f File) WriteYAML(in interface{}) error {
+	encoded, err := yaml.Marshal(in)
+	if err != nil {
+		return err
+	}
+	return f.Write(string(encoded))
+}
+
+// The following files configure and authenticate the coder-cli in ~/.coder/.
```

Comment:
nit: The `.` at the end of this sentence made me think all config files would be dotfiles.
I think linting complains if you don't end in a period, maybe `... coder-cli in '~/.coder/'.`